File: 17_factorial.py
# pyhton program to find the factorial of a nunmber

# mehtod 1 - by using simple iteration
try:
    n=int(input("Enter a number :"))
    fac=1
    if n<0:
        print("Enter a value greater than or equal to ZERO")
    else:
        for i in range(1,n+1):
            fac=fac*i
        print(fac)
except:
    print("Enter a valid integer value!")

# method 2 - using recurssion 
try:
    n=int(input("Enter a number :"))
    if n<0:
        print("Enter a value greater than or equal to ZERO")
    else:
        def fac(a):
            if a==0:
                return 1
            return a*fac(a-1)
        print(fac(n))
except:
    print("Enter a valid integer value.")

File: 17_factorial/17_factorial.dart
import 'dart:io';

int factorialValue(int num) {
  int fac = 0;

  if (num == 0) {
    return 1;
  } else {
    for (int i = 1; i <= num; i++) {
      fac = fac * i;
    }
    return fac;
  }
}

void main() {
  try {
    while (true) {
      stdout.write("Enter a positive integer value : ");
      int num = int.parse(stdin.readLineSync()!);

      if (num < 0) {
        print("Enter a value greater than or equal to ZERO");
      } else {
        int result = factorialValue(num);
        print(result);
      }

      stdout.write("Do you want to continue? (y/n) : ");
      String choice = stdin.readLineSync()!.toLowerCase();
      if (choice == 'n') {
        print("Program Finished");
        break;
      }
    }
  } on FormatException catch (e) {
    print("Invalid input : $e");
  } on Exception catch (e) {
    print("Exception caught : $e");
  }
}
